### ISSUE TITLE
old_log_probs VS ref_log_probs

### ISSUE THREAD

**Issue by Nipers** [PRIME-RL/PRIME#45]
According to the pseudo code in the blog, the reference model is always the SFT version model, I found this code in the DPPrime.py:
```
if self.reference_module is not None:
    ref_log_prob = torch.cat([self._forward_micro_batch(module=self.reference_module, micro_batch=data[i:i + 1], prompt_length=prompt_length, no_grad=True) for i in range(len(data))])
else:
    ref_log_prob = data['old_log_probs'] 
```
and this code in the DPActor.py:
```
old_log_prob = data['old_log_probs']
advantages = data['advantages']

logits, log_prob = self._forward_micro_batch(micro_batch=data, temperature=temperature)

pg_loss, pg_clipfrac, ppo_kl = core_algos.compute_policy_loss(old_log_prob=old_log_prob,
                                                              log_prob=log_prob,
                                                              advantages=advantages,
                                                              eos_mask=response_mask,
                                                              cliprange=clip_ratio)
```
My question is, when is the `old_log_probs` set as the `ref_log_prob` if we set the SFT model as reference model and `ref_log_prob` should be used to calculate the PPO loss for the policy model.
I have tried to search for the setting operation but found nothing, does it mean that the PPO loss is always calculated by the old policy probs rather than the reference probs? 
Looking forward to your reply.

**Comment by cgq15** [comment]
oh that was a mistake in pseudo code. \pi_ref in PPO loss always refers to old_log_prob of policy model, not the ref model of PRM. Sorry for the mistake, will fix it soon!

**Comment by Nipers** [comment]
Thank you very much!
